actgen/utils.py

```diff
@@ -4,12 +4,104 @@
 import logging
 from pydoc import locate
 
+import gym
+import argparse
 import torch
-from matplotlib import pyplot as plt
+
+from . import wrappers as wrap
 
 Experience = namedtuple('Experience', ['state', 'action', 'reward', 'next_state', 'done'])
 
 
+class Trial:
+    """
+    a base trial class for running experiments
+    """
+    def __init__(self):
+        pass
+
+    def parse_common_args(self):
+        parser = argparse.ArgumentParser(
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            add_help=False
+        )
+        # common args
+        parser.add_argument('--env_name', type=str, default='CartPole-v0',
+                            help='Which gym environment to use (abbreviate Atari envs: e.g. "MsPacman")')
+        parser.add_argument('--agent', type=str, default='dqn',
+                            choices=['dqn', 'random', 'action_dqn'],
+                            help='Which agent to use')
+        parser.add_argument('--duplicate', '-d', type=int, default=5,
+                            help='Number of times to duplicate actions')
+        parser.add_argument('--random_actions', default=False, action='store_true',
+                            help='Make the duplicate actions all random actions')
+        parser.add_argument('--seed', '-s', type=int, default=0,
+                            help='Random seed')
+        parser.add_argument('--hyperparams', type=str, default='hyperparams/defaults.csv',
+                            help='Path to hyperparameters csv file')
+        parser.add_argument('--test', default=False, action='store_true',
+                            help='Enable test mode for quickly checking configuration works')
+        parser.add_argument('--results_dir', type=str, default='./results/',
+                            help='Path to the result directory to save model files')
+        parser.add_argument('--tag', type=str, default='default_exp',
+                            help='A tag for the current experiment, used as a subdirectory name for saving models')
+        parser.add_argument('--disable_gpu', default=False, action='store_true',
+                            help='enforce training on CPU')
+        return parser
+
+    def parse_unknown_args(self, parser):
```

Comment:
I find this function name confusing as well, since it parses both known and unknown args, and specifically returns the known args (which happen to also contain any unknown ones). How about just `parse_args()`?